### ISSUE TITLE
weight renorm in softmax implementation

### ISSUE THREAD

**Issue by ZhengRui** [Paralysis/ringloss#3]
should this line 
https://github.com/Paralysis/ringloss/blob/master/loss_func.py#L19
be:
`self.fc.weight.data.renorm_(2, 0, 1e-5).mul_(1e5)` ?

**Comment by Paralysis** [comment]
@ZhengRui PyTorch's FC layer weight is defined as:
`self.weight = Parameter(torch.Tensor(out_features, in_features))`
So dimension 1 should be normalized, i.e. normalize the weight vector for each class.

**Comment by ZhengRui** [comment]
yes we should normalize the weight vector for each class, but when using `torch.renorm(input, p, dim, maxnorm)`, `dim` should be 0 to actually normalized each row

**Comment by Paralysis** [comment]
Fixed! Thanks for correction!
